internal/disk/path_trie.go

```diff
@@ -0,0 +1,113 @@
+package disk
+
+import (
+	"sort"
+	"strings"
+)
+
+func pathTrieSplitPath(path string) []string {
+	path = strings.Trim(path, "/")
+	if path == "" {
+		return []string{}
+	}
+
+	return strings.Split(path, "/")
+}
+
+type PathTrie struct {
+	Name    []string
```

Comment:
For:
```go
map[string]interface{}{
	"/":                            common.IntToPtr(1),
	"/var":                         common.IntToPtr(2),
	"/var/lib/chrony":              common.IntToPtr(7),
	"/var/lib/osbuild":             common.IntToPtr(3),
	"/var/lib/osbuild/store/cache": common.IntToPtr(4),
	"/boot":                        common.IntToPtr(5),
	"/boot/efi":                    common.IntToPtr(6),
}
```
you'll get
```go
PathTrie{
	Name:    []string{},
	Payload: common.IntToPtr(1),
	Paths: []*PathTrie{
		{
			Name:    []string{"boot"},
			Payload: common.IntToPtr(5),
			Paths: []*PathTrie{
				{
					Name:    []string{"efi"},
					Payload: common.IntToPtr(6),
				},
			},
		},
		{
			Name:    []string{"var"},
			Payload: common.IntToPtr(2),
			Paths: []*PathTrie{
				{
					Name:    []string{"lib", "chrony"},
					Payload: common.IntToPtr(7),
					Paths: []*PathTrie{
						{
							Name:    []string{"osbuild"},
							Payload: common.IntToPtr(3),
							Paths: []*PathTrie{
								{
									Name:    []string{"store", "cache"},
									Payload: common.IntToPtr(4),
								},
							},
						},
					},
				},
			},
		},
	},
}
```
which does not look correct.

Reply:
> Is there any particular reason why you didn't choose to implement the trie with a single directory being represented by a single node?

Ironically, that was the first version but I didn't like the un-needed intermediate nodes; and it made it impossible to distinguish between "helper" nodes and actual nodes. This is currently not important but I had a version with a better error message (with the help of a `fullname` field) that would show you which exact policy is preventing the mountpoint.

Reply:
> However there is a corner case, in which it will produce unexpected results.

Yeah, I thought we would not need that special case, but you are of course right that it should at least be documented. On a second thought, maybe I should just fix the implementation because it is weird to have that limitation even though it will make the code more complicated.

Reply:
Reimplementation and preventing hitting this case sounds better to me than documenting this "feature". Thanks a lot!